src/lib/operators/table_scan/column_between_table_scan_impl.cpp

```diff
@@ -37,17 +37,28 @@ std::string ColumnBetweenTableScanImpl::description() const { return "ColumnBetw
 void ColumnBetweenTableScanImpl::_scan_non_reference_segment(
     const BaseSegment& segment, const ChunkID chunk_id, RowIDPosList& matches,
     const std::shared_ptr<const AbstractPosList>& position_filter) const {
-  const auto ordered_by = _in_table->get_chunk(chunk_id)->ordered_by();
-  if (ordered_by && ordered_by->first == _column_id) {
-    _scan_sorted_segment(segment, chunk_id, matches, position_filter, ordered_by->second);
-  } else {
-    // Select optimized or generic scanning implementation based on segment type
-    if (const auto* dictionary_segment = dynamic_cast<const BaseDictionarySegment*>(&segment)) {
-      _scan_dictionary_segment(*dictionary_segment, chunk_id, matches, position_filter);
-    } else {
-      _scan_generic_segment(segment, chunk_id, matches, position_filter);
+  const auto chunk_sorted_by = _in_table->get_chunk(chunk_id)->sorted_by();
+
+  const auto* dictionary_segment = dynamic_cast<const BaseDictionarySegment*>(&segment);
+  if (chunk_sorted_by &&
+      !(dictionary_segment && position_filter && _in_table->column_data_type(_column_id) == DataType::String)) {
```

Comment:
Blocking myself here. There are still some performance regressions remaining. Will have a look at them. Further, I'll take a look at @mrks's issue with the aggregate sort.